src/transformer/genericDeclaration/genericDeclaration.ts

```diff
@@ -110,10 +167,15 @@ export function GenericDeclaration(scope: Scope): IGenericDeclaration {
           }
         }
 
+        if (!typeParameterDeclaration || !scope.isBound()) {
```

Comment:
It may be a dumb question but I still don't understand why the previous `if statement` is not good enough to make sure is a circular generic dependency?

I mean this ```const isExtendingItself: boolean = MockDefiner.instance.getDeclarationKeyMap(typeParameterDeclaration) === declarationKey;```

Reply:
The old condition was fine if you were to completely ignore circular generics and terminate early.

However, this time around, we want to process the declaration exactly twice, where the second iteration avoids calling `GetDescriptor` (which is what's causing the infinite recursion) and instead utilizes `this.constructor` for a back-reference.

`isExtendingItself` does not help us in that case if we were to use it to our advantage, since the type checker would bring us to the same declaration on every `GetDescriptor` call and result in the same scenario, yes, it is extending itself, over and over.